src/rgw/rgw_object_lock.cc

```diff
@@ -5,6 +5,15 @@
 
 using namespace std;
 
+void DefaultRetention::decode_json(JSONObj *obj) {
+  JSONDecoder::decode_json("mode", mode, obj);
+  if (days > 0) {
+    JSONDecoder::decode_json("days", days, obj);
```

Comment:
this doesn't look right. `(days > 0)` depends on a value that we haven't decoded yet

`DefaultRetention::decode_xml()` just below tries to decode both "days" and "years", but complains if it finds neither/both

Reply:
thanks. done.